pex/bin/pex.py

```diff
@@ -525,6 +530,28 @@ def build_pex(args, options, resolver_option_builder):
       for interpreter in options.python or [None]
     ]
 
+  # NB: this block is neccesary to load variables from the pexrc runtime configuration.
+  # It is only neccesary to cover the case where the pexrc file is located in the same
+  # directory as the output pex. The other supported pexrc locations (/etc/pexrc) do not
+  # require special treatment for reading at build time.
+  if options.pex_name:
+    if '/' in options.pex_name:  # the output pex is not in the working directory
+      pexrc_dir = os.path.dirname(options.pex_name)
+    else:
+      pexrc_dir = os.getcwd()
+  else:
```

Comment:
So making this change actually broke the ability to read from the pexrc at runtime. Seems like the author expects .pexrc to be in the same directory as the pex that is running, which makes sense to me, at least for directory-level configurations. 

The runtime breakage has to do with os.path.dirname(sys.argv[0]) being the location of the running pex which is different than os.getcwd() at runtime (the root of the pex project I am running these tests in).  I think this could be a quirk of running all of this in pytest, however it might make sense to leave it as is because if we change to getcwd() it will require users to either have a .pexrc from wherever they decide to execute a pex from (which could be annoyling/problematic if absolute/long relative paths are used), or always run a pex from the directory the pex is in, with the expectation that the .pexrc is in it. 

Given that getcwd() broke the runtime config reading, I think I should leave it as is and make an issue + TODO to circle back, as I think this is beyond the scope of these changes and is more specific to the case where .pexrc is used. Thoughts?

Reply:
makes sense. I think whatever the case tho, this block of code here:

```
+  # NB: this block is necessary to load variables from the pexrc runtime configuration.
 +  # It is only necessary to cover the case where the pexrc file is located in the same
 +  # directory as the output pex. The other supported pexrc locations (/etc/pexrc) do not
 +  # require special treatment for reading at build time.
 +  if options.pex_name:
 +    if '/' in options.pex_name:  # the output pex is not in the working directory
 +      pexrc_dir = os.path.dirname(options.pex_name)
 +    else:
 +      pexrc_dir = os.getcwd()
 +    pexrc = os.path.join(pexrc_dir, '.pexrc')
 +  else:
 +    pexrc = '.pexrc'
```

feels pretty hacky and like it's largely intended to make tests pass. to me, this sort of logic belongs in `Variables` instead so that the behavior of PP/PPP discovery is uniform across all callers. this block also doesn't consider `~/.pexrc` like `Variables` as used today does and will look at `.pexrc` 2x in the case that `options.pex_name` isn't defined. I'm also not quite sure why we would consider the `.pexrc` relative to a new output path.

so as long as we can clean that aspect up, I don't feel strongly about `os.getcwd()` vs `os.path.dirname(sys.argv[0])`.

Reply:
Ok. I made some changes that I think might be along the line of what you are thinking of, however please let me know if theres a better way to do this. This was as simple as I could manage. 

The .pexrc relative to output path is for when you have an absolute path to the output pex specified for the `-o /path/to/pex/pex` flag of the PEX CLI. This hacky-ness of this is largely due to the fact that I am trying to process rc files that are intended to be used at runtime only. This grew out of the existing integration tests placing the pexrc files in the same directory as the output pex via `os.path.join`. I think it still makes sense to have this in place for the case where the output pex is not the cwd.

Anyhow, I took all of that out of bin/pex.py and moved it into `Variables`.